core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -473,7 +473,7 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
 
     // Validate spark.executor.extraJavaOptions
     getOption(executorOptsKey).foreach { javaOpts =>
-      if (javaOpts.contains("-Dspark")) {
+      if (javaOpts.contains("-Dspark") && !javaOpts.contains("-Dspark.hadoop.")) {
```

Comment:
And this part should be eventually merged with #445, that supports extra JVM options for executors.